YouTubeCrawler/src/main/scala/com/sofiia/production/akka/CollectionHandlerActor.scala

```diff
@@ -0,0 +1,17 @@
+package com.sofiia.production.akka
+
+import akka.actor.Actor
+import com.sofiia.production.functions.io.FileWriterCustom
+
+import scala.collection.mutable.ArrayBuffer
+
+class CollectionHandlerActor(array: ArrayBuffer[String], expectedSize: Int,  fileNameForAkka: String) extends Actor {
```

Comment:
expected size - is not good architecture decision. What would happen if smth fails? Let's make also Timer actor which will by time send msg to this one and your actor will write everything in storage to file. But, it's also necessary to prevent semistate situation when you are editing stage collection during write procedure. Find a way to implement this logic)